lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -781,6 +784,96 @@ def get(self, request, *args, **kwargs):
         return render_to_response(self.template_name, context)
 
 
+class MgmtCommands(SysadminDashboardView):
+    """This class provides a view to handle the execution of management commands"""
+
+    def get(self, request):
+        """Displays the view for the management commands tab"""
+
+        if not request.user.is_staff:
+            raise Http404
+
+        context = {
+            'msg': self.msg,
+            'djangopid': os.getpid(),
+            'modeflag': {'mgmt_commands': 'active-section'},
+            'edx_platform_version': getattr(settings, 'EDX_PLATFORM_VERSION_STRING', ''),
+        }
+        return render_to_response(self.template_name, context)
+
+
+class ExecuteCommand(SysadminDashboardView):
+    """this class handles the execution of any management commands via ajax"""
+
+    def execute_command_and_capture_output(self, command, *args, **kwargs):
+        """
+        Executes the specified command with the given args and kwargs, and returns the console output
+        :param command:
+        :param args:
+        :param kwargs:
+        :return: client_stdout, client_stderr, client_error
+        """
+        orig_stdout = sys.stdout
+        orig_stderr = sys.stderr
+
+        sys.stdout = stdout_buffer = StringIO.StringIO()
+        sys.stderr = stderr_buffer = StringIO.StringIO()
+
+        client_error = None
+        try:
+            call_command(command, *args, **kwargs)
+        except SystemExit:
+            client_error = _('Command failed')
+        except Exception as error:
+            client_error = error
+
+        sys.stdout = orig_stdout
+        sys.stderr = orig_stderr
+        stdout_buffer.seek(0)
+        stderr_buffer.seek(0)
+
+        client_stdout = stdout_buffer.read()
+        client_stderr = stderr_buffer.read()
+        return client_stdout, client_stderr, client_error
+
+    def parse_parameters(self, params):
+        """
+        Parses the list of parameters from the ExecuteCommand post request
+        :param params:
+        :return: command_name, kwargs, args
+        """
+        command_name = params.pop('command')[0]
+
+        kwargs = {}
+        args = []
+        # collect kwargs and args
+        if 'args' in params:
```

Comment:
No, because when getting the args from the parameters hash, we also want to remove the args from parameters, hence the use of pop. The structure of parameters is {'args': [], 'kwflags':[], 'kwarg1': 'value1', 'kwarg2': value2...}